i7aof/biascorr/projection.py

```diff
@@ -0,0 +1,372 @@
+import os
+from glob import glob
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from tqdm import tqdm
+
+from i7aof.biascorr.timeslice import Timeslice
+
+
+def status(process_name):
+    def inner_decorator(func):
+        def wrapper(*args, **kwargs):
+            statement = process_name
+            for _key, value in kwargs.items():
+                statement += f'{value}'
+
+            Ndots = max(3, 40 - len(statement))
+            print(f'{statement} {"." * Ndots} \033[033mRunning\033[0m')
+            result = func(*args, **kwargs)
+            # Code to execute after the function call
+            print(f'\033[F{statement} {"." * Ndots} \033[032mFinished\033[0m')
+            return result
+
+        return wrapper
+
+    return inner_decorator
+
+
+class Projection:
+    """
+    A main class for projections
+
+    Attributes
+    ----------
+    config : mpas_tools.config.MpasConfigParser
+        Configuration options.
+
+    logger : logging.Logger
+        Logger for the class.
```

Comment:
Like for the TimeSlice class, it would be good to add more attributes here.  Looking through, these seem like the important ones:
```
ref
modref
base
basinmask
basinNumber
Tscaling
```